pkg/compressio/nocompressio.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2023 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package compressio also provides non-compressing IO interface, as well
```

Comment:
While this comment is useful, godoc doesn't deal well with multiple package comments spread across different files.

I suggest expanding the package comment in `compressio.go` that states that the package also provides a no-compression variant, and then move the comment in `nocompressio.go` somewhere else in the file (maybe after the imports, or maybe as part of the documentation for `SimpleReader`) so that it's not a package comment.

Reply:
To be clear, I am only referring to the part of the comment that starts with "// Package compressio also provides...". The copyright notice should remain where it is.

Reply:
I rewrote the package comment as follows:

```
  // Package compressio provides parallel compression and decompression, as well
- // as optional SHA-256 hashing.
+ // as optional SHA-256 hashing. It also provides another storage variant
+ // (nocompressio) that does not compress data but tracks its integrity.
```